File: 프로그래머스/2/42587. 프로세스/프로세스.cs
using System;
using System.Collections.Generic;
using System.Linq; 

public class Solution {
    public int solution(int[] priorities, int location) {
        Queue<(int idx, int priority)> q = new Queue<(int,int)>();
        for(int i = 0; i < priorities.Length; i++)
        {
            q.Enqueue((i, priorities[i]));
        }
        
        int answer = 0;
        
        while (q.Count > 0)
        {
            var process = q.Dequeue();
            
            if (q.Any(x=> x.priority > process.priority))
            {
                q.Enqueue(process);
            }
            else
            {
                answer++;
                
                if(process.idx == location)
                {
                    return answer;
                }
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    answer = 0
    process = deque([(p,i) for i, p in enumerate(priorities)])
    
    while process:
        q = process.popleft()
        
        if any(q[0] < p[0] for p in process):
            process.append(q)
        else:
            answer += 1
            
            if q[1] == location:
                return answer